pkg/cloudprovider/providers/aws/instances.go

```diff
@@ -76,8 +77,8 @@ func (name kubernetesInstanceID) mapToAWSInstanceID() (awsInstanceID, error) {
 
 	// We sanity check the resulting volume; the two known formats are
 	// i-12345678 and i-12345678abcdef01
-	// TODO: Regex match?
-	if awsID == "" || strings.Contains(awsID, "/") || !strings.HasPrefix(awsID, "i-") {
+	reg := regexp.MustCompile("^i-[^/]*$")
```

Comment:
the resulting `reg` could be made into a global variable, so it doesn't need to be compiled on every pass.

Reply:
I have thought about it , but get grid of it after consider readability of code. I will fix it, Tahnks @rphillips

Reply:
Bad news, the `regexp.MustCompile("^vol-[^/]*$")` can't initializer as const. @rphillips